src/drive/payments/object-storage.ts

```diff
@@ -0,0 +1,81 @@
+import { ApiUrl, AppDetails } from '../../shared';
+import { basicHeaders } from '../../shared/headers';
+import { HttpClient } from '../../shared/http/client';
+import { CreatedSubscriptionData } from './types';
+
+interface ObjectStoragePlan {
+  id: string;
+  bytes: number;
+  interval: 'year' | 'month' | 'lifetime';
+  amount: number;
+  currency: string;
+  decimalAmount: number;
+}
+
+export class ObjectStorage {
+  private readonly client: HttpClient;
+  private readonly appDetails: AppDetails;
+
+  public static client(apiUrl: ApiUrl, appDetails: AppDetails) {
+    return new ObjectStorage(apiUrl, appDetails);
+  }
+
+  private constructor(apiUrl: ApiUrl, appDetails: AppDetails) {
+    this.client = HttpClient.create(apiUrl);
+    this.appDetails = appDetails;
+  }
+
+  public getObjectStoragePlanById(priceId: string, currency?: string): Promise<ObjectStoragePlan> {
+    const query = new URLSearchParams();
+    priceId !== undefined && query.set('planId', priceId);
+    currency !== undefined && query.set('currency', currency);
+    return this.client.get(`/object-storage-plan-by-id?${query.toString()}`, this.headers());
+  }
+
+  public createCustomerForObjectStorage(
+    name: string,
+    email: string,
+    country?: string,
+    companyVatId?: string,
+  ): Promise<{ customerId: string; token: string }> {
+    return this.client.post(
+      '/create-customer-for-object-storage',
+      {
+        name,
+        email,
+        country,
+        companyVatId,
+      },
+      this.headers(),
+    );
+  }
+
+  public createObjectStorageSubscription(
+    customerId: string,
+    plan: ObjectStoragePlan,
+    token: string,
+    companyName: string,
+    vatId: string,
+    promoCodeId?: string,
+  ): Promise<CreatedSubscriptionData> {
```

Comment:
as there are too many arguments of the same type, add it to an object, so we reduce the possibility of passing the wrong arguments to the function